src/vcpkg/base/downloads.cpp

```diff
@@ -974,12 +980,15 @@ namespace vcpkg
                     fs, urls, headers, download_path, sha512, m_config.m_secrets, errors, progress_sink);
                 if (auto url = maybe_url.get())
                 {
+                    m_config.m_read_url_template.has_value() ? msg::println(msgAssetCacheMiss, msg::url = urls[0])
+                                                             : msg::println(msgDownloadingUrl, msg::url = urls[0]);
+
                     if (auto hash = sha512.get())
                     {
                         auto maybe_push = put_file_to_mirror(fs, download_path, *hash);
                         if (!maybe_push)
                         {
-                            msg::println_warning(msgFailedToStoreBackToMirror);
+                            msg::println_warning(msgFailedToStoreBackToMirror, msg::path = download_path.filename());
```

Comment:
It's not clear to me what clarity showing the file name here adds to the situation. We already told the user what the file name was. What the user might actually care about is what "mirror" means because I don't believe we use that term anywhere else.